### PR TITLE
bugfix/56635 - fix spacing errors caused by unhandled characters

### PR DESCRIPTION
**Pull request by kais-sapper**
bugfix/56635 - fix spacing errors caused by unhandled characters

- Move `he.decode` to `normalizeMarkdownInput` instead of performing it on a per-token basis
- Introduce and use `INVISIBLE_FORMAT_AND_CONTROL_RE` to strip control, format, surrogate, and replacement characters during `normalizeMarkdownInput`
- Improve script detection and font routing in `md-to-pdf/jspdfFonts.ts` so symbol/combining/non-latin edge cases resolve to NotoSerif instead of latin fallback
- Update tests to match new behaviour

### CODE REVIEW

**Review by braydonc21 — APPROVED** [review]
All looks good

File: md-to-pdf/jspdfFonts.ts [hunk 1]
```diff
@@ -114,6 +120,7 @@ export function chooseFontForText(text: string, fallback = "helvetica"): string
         case "korean": return "NotoSerifKorean"; // covers Hangul
         case "japanese": return "NotoSerifJapanese"; // covers Kana
         case "emoji": return "NotoSerif"; // or handle as images
+        case "symbol": return "NotoSerif";
```
Comment (divya-b-shiva) [inline md-to-pdf/jspdfFonts.ts:123]: since we are handling the different types here, can we remove the unicode 32 requirement from hunchly?

**Review by divya-b-shiva — APPROVED** [review]
@kais-sapper I just want to check, if this library handles unicode 32 characters, can we remove that from hunchly?
